pkg/nfd-client/topology-updater/nfd-topology-updater.go

```diff
@@ -209,59 +170,41 @@ func (w *nfdTopologyUpdater) Stop() {
 	}
 }
 
-// connect creates a client connection to the NFD master
-func (w *nfdTopologyUpdater) Connect() error {
-	// Return a dummy connection in case of dry-run
-	if w.args.NoPublish {
-		return nil
-	}
+func nodeName() string { return os.Getenv("NODE_NAME") }
 
-	if err := w.NfdBaseClient.Connect(); err != nil {
+func (w *nfdTopologyUpdater) updateNodeResourceTopology(zoneInfo v1alpha1.ZoneList) error {
+	cli, err := w.apihelper.GetTopologyClient()
+	if err != nil {
 		return err
 	}
-	w.client = pb.NewNodeTopologyClient(w.ClientConn())
-
-	return nil
-}
 
-// disconnect closes the connection to NFD master
-func (w *nfdTopologyUpdater) Disconnect() {
-	w.NfdBaseClient.Disconnect()
-	w.client = nil
-}
-
-// advertiseNodeTopology advertises the topology CR to a Kubernetes node
-// via the NFD server.
-func advertiseNodeTopology(client pb.NodeTopologyClient, zoneInfo v1alpha1.ZoneList, tmPolicy string, nodeName string) error {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-	zones := make([]*v1alpha1.Zone, len(zoneInfo))
-	// TODO: Avoid copying of data to allow returning the zone info
-	// directly in a compatible data type (i.e. []*v1alpha1.Zone).
-	for i, zone := range zoneInfo {
-		zones[i] = &v1alpha1.Zone{
-			Name:      zone.Name,
-			Type:      zone.Type,
-			Parent:    zone.Parent,
-			Resources: zone.Resources,
-			Costs:     zone.Costs,
+	nrt, err := cli.TopologyV1alpha1().NodeResourceTopologies().Get(context.TODO(), nodeName(), metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		nrtNew := v1alpha1.NodeResourceTopology{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nodeName(),
+			},
+			Zones:            zoneInfo,
+			TopologyPolicies: []string{w.nodeInfo.tmPolicy},
 		}
-	}
 
-	topologyReq := &pb.NodeTopologyRequest{
-		Zones:            zones,
-		NfdVersion:       version.Get(),
-		NodeName:         nodeName,
-		TopologyPolicies: []string{tmPolicy},
+		_, err := cli.TopologyV1alpha1().NodeResourceTopologies().Create(context.TODO(), &nrtNew, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to create v1alpha1.NodeResourceTopology!:%w", err)
+		}
+		return nil
+	} else if err != nil {
+		return err
 	}
 
-	utils.KlogDump(1, "Sending NodeTopologyRequest to nfd-master:", "  ", topologyReq)
+	nrtMutated := nrt.DeepCopy()
+	nrtMutated.Zones = zoneInfo
 
-	_, err := client.UpdateNodeTopology(ctx, topologyReq)
+	nrtUpdated, err := cli.TopologyV1alpha1().NodeResourceTopologies().Update(context.TODO(), nrtMutated, metav1.UpdateOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to update v1alpha1.NodeResourceTopology!:%w", err)
 	}
-
+	utils.KlogDump(2, "CR instance updated resTopo:", "  ", nrtUpdated)
```

Comment:
I would increase the verbosity to at least 3.
On a cluster with a high update rate, it might spam the log a little bit too much.

Reply:
I was basically only movie/refactor code, trying to not change any functionality. But I guess this makes sense -> change log level to 4